product/02-innersource-and-agile.md

```diff
@@ -0,0 +1,82 @@
+# InnerSource and Agile
+
+You want to improve your product and deliver those improvements faster to customers.
+You want to make stakeholders happy.
+InnerSource helps your team deliver value and maintain autonomy especially in a highly interconnected world.
+
+
+## Autonomous teams in an interconnected world
+
+* organisations try to deliver value to customers quickly
+* to do that lengthy roadmap synchronisation settings are to be avoided
+* often those discussions stem from team dependencies: Releases waiting for the testers, deployment waiting for time in the ops team.
+As a result, cross functional teams where testing happens close to development and those who build software own running the software have proven to be successful.
+* but there is another type of inter-team dependency:
+* however we live in an interconnected world: Software written today relies on many dependencies
+* Each source code dependency, each call to some service, each external dataset used boils down to a dependency between two teams.
+* What if you need changes made to such a dependency?
+   * Traditional options:
+      * submit a feature request and wait it out
+      * build a workaround
+      * if nothing else helps, escalate your issue
+   * added benefit of InnerSource:
+      * help with the solution
+
+* But team members working in each others repositories sounds like chaos?
+   * InnerSource comes with a set of roles and processes, contributions happen in a structured way
+   * each InnerSource project has a clear set of trusted committers with clear accountabilities that go beyond simply reviewing code.
+   * support for more autonomy by being able to fix upstream artifacts without sacrifycing quality due to a clear review process
+
+* Todays world moves towards a remote first culture - InnerSource has a focus on written communication and lends itself easily to remote first.
+
+
+## Does InnerSource replace Agile?
+
+* short answer: No, not at all.
+* Instead they complement each other:
+
+
+- How InnerSource and Agile complement each other
+   - Well factored and well tested code leads to faster on-ramp for agile teams, but also helps InnerSource allowing for faster on-ramp for contributors
+   - A focus on working together and not assigning individual tasks allows for easier on-ramp of contributors.
+   - Existing local pairing can make cross team pairing and sharing of work in progress easier.
+   - Existing test driven development means that confidence in contributions is higher.
+   - XP teams have a "leave the code in better shape than you found it" attitude - mitigates the risk of decline in quality or co-hesion of code when faced with multiple contributions from different sources.
+
+## Common misunderstandings when coming from agile teams
+
+* Impact on performance reviews: InnerSource moves some of the energy you spend developing outside of your own team.
```

Comment:
Re-reading that bullet point now, it's in the wrong section.

In Germany at least companies have regular, often once a year, or quarterly or something like that, performance reviews where your line manager looks at what you accomplished and what your focus area should be in the coming months. Typically that manager is fairly close to your daily work so it's easy to talk about the accomplishments of previous weeks and months. As employees move to roles that have impact beyond their local teams that conversation needs to adjust. InnerSource is a bit special here: Contributors and Trusted Committers by their job title come from a "software engineer" background, but their impact may go across several hierarchical branches inside of the organisation - though a bit less than in Open Source: If in Open Source you work on backstage, foundation level topics, your work may impact even other corporations, but still be invisible - because private - to the person who agreed to donate your time to that foundation. In InnerSource all that remains within one corporation, but might still be out of sight for where your manager and your team typically operates.

Reply:
Ultimately there need to be factors that go into KPIs. 

Contributing outside of your team and mentoring contributors is something that I would expect to become part of seniority levels.

If we look back to the "enlightened self interest" for open source contributions I could see something similar for InnerSource as well: 
* Move faster by standing on the shoulders of giants, combined with
* accomplish more by collaborating across business units (like we do here for the Learning Path segment - we all need it, it's not a unique selling point for neither of our employers, but by collaborating we can create something that would be more costly when done internally only), combined with
* fix issues that block shipping features to your customers and move maintenance to a team that is larger than your local team
In advanced settings I would expect people to understand the value of contributors working on simpler features freeing the host team to work on more complex changes that the contributors have a business need for.

Reply:
Essentially I would like to capture some of the learnings that Myrle Krantz shared in her keynote at the InnerSource summit in that segment.

Reply:
Is there any update to make to the outline to incorporate these ideas?

Reply:
Good point - I'll see where these points fit.